linking error with gcc 10

#### Expected behaviour
Succesfull build with gcc 10

#### Actual behaviour
Linking error when building against gcc-10.0.1-0.6.fc32 at fedora rawhide (next release) branch

#### Steps to reproduce the behaviour
See build log at https://copr-be.cloud.fedoraproject.org/results/raveit65/MATE-1.23.x/fedora-rawhide-x86_64/01207271-mate-utils/build.log.gz
```
libtool: link: gcc -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -pthread -I/usr/include/libgtop-2.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -Wall -Wmissing-prototypes -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wl,-z -Wl,relro -Wl,--as-needed -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o mate-disk-usage-analyzer mate_disk_usage_analyzer-baobab.o mate_disk_usage_analyzer-baobab-cell-renderer-progress.o mate_disk_usage_analyzer-baobab-ringschart.o mate_disk_usage_analyzer-baobab-scan.o mate_disk_usage_analyzer-baobab-treeview.o mate_disk_usage_analyzer-baobab-utils.o mate_disk_usage_analyzer-callbacks.o mate_disk_usage_analyzer-baobab-prefs.o mate_disk_usage_analyzer-baobab-remote-connect-dialog.o mate_disk_usage_analyzer-baobab-chart.o mate_disk_usage_analyzer-baobab-treemap.o mate_disk_usage_analyzer-baobab-resources.o -Wl,--export-dynamic  -lm -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgtop-2.0 -lglib-2.0 -pthread
/usr/bin/ld: mate_disk_usage_analyzer-baobab-scan.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
/usr/bin/ld: mate_disk_usage_analyzer-baobab-treeview.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
/usr/bin/ld: mate_disk_usage_analyzer-baobab-utils.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
/usr/bin/ld: mate_disk_usage_analyzer-callbacks.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
/usr/bin/ld: mate_disk_usage_analyzer-baobab-prefs.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
/usr/bin/ld: mate_disk_usage_analyzer-baobab-remote-connect-dialog.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: multiple definition of `baobab'; mate_disk_usage_analyzer-baobab.o:/builddir/build/BUILD/mate-utils-1.23.2/baobab/src/baobab.h:102: first defined here
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:559: mate-disk-usage-analyzer] Error 1
make[4]: Leaving directory '/builddir/build/BUILD/mate-utils-1.23.2/baobab/src'
make[3]: *** [Makefile:474: all] Error 2
make[3]: Leaving directory '/builddir/build/BUILD/mate-utils-1.23.2/baobab/src'
make[2]: *** [Makefile:438: all-recursive] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/mate-utils-1.23.2/baobab'
make[1]: *** [Makefile:504: all-recursive] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/mate-utils-1.23.2'
make: *** [Makefile:436: all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.DyrUDw (%build)
RPM build errors:
    extra tokens at the end of %else directive in line 124:  %else # 0%{?rel_build}
    extra tokens at the end of %endif directive in line 128:  %endif # 0%{?rel_build}
    Bad exit status from /var/tmp/rpm-tmp.DyrUDw (%build)
Child return code was: 1
EXCEPTION: [Error()]
```
This is probably caused by new default `-fno-common` with gcc-10, see
https://gcc.gnu.org/gcc-10/porting_to.html#common
```
Default to -fno-common

A common mistake in C is omitting extern when declaring a global variable in a header file. If the header is included by several files it results in multiple definitions of the same variable. In previous GCC versions this error is ignored. GCC 10 defaults to -fno-common, which means a linker error will now be reported. To fix this, use extern in header files when declaring global variables, and ensure each global is defined in exactly one C file. As a workaround, legacy C code can be compiled with -fcommon.


      int x;  // tentative definition - avoid in header files

      extern int y;  // correct declaration in a header file
  
```
#### MATE general version
1.23.x

#### Package version
mate-utils-1.23.2

#### Linux Distribution
Fedora rawhide

#### Link to downstream report of your Distribution
1.23.2 isn't build at the moment.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/87736167-linking-error-with-gcc-10?utm_campaign=plugin&utm_content=tracker%2F1509859&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1509859&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/baobab/src/baobab.c
+++ b/baobab/src/baobab.c
@@ -42,6 +42,8 @@
 static void push_iter_in_stack (GtkTreeIter *);
 static GtkTreeIter pop_iter_from_stack (void);
 
+BaobabApplication baobab;
+
 static gint currentdepth = 0;
 static GtkTreeIter currentiter;
 static GtkTreeIter firstiter;

--- a/baobab/src/baobab.h
+++ b/baobab/src/baobab.h
@@ -99,7 +99,7 @@ struct _BaobabApplication {
 };
 
 /* Application singleton */
-BaobabApplication baobab;
+extern BaobabApplication baobab;
 
 struct chan_data {
 	guint64 size;

